### PR TITLE
added beginner tutorial for linear regression

### PR DESCRIPTION
**Pull request by jsnowdon99**

### CODE REVIEW

**Comment by svekars** [comment]
@jsnowdon99 thanks for contributing to PyTorch tutorials! Can we please move this to the recipes/ as this is more of a bite-size content? Also, one suggestion to make it more educational is to break the single code block into multiple code blocks with descriptions on what's happening in each block?

File: beginner_source/linear_regression_tutorial.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+================
+**Author:** Jack Snowdon
+
+In this tutorial, we will generate a linear regression with PyTorch to find
+the best-fitting line in a linear relationship between an independent and 
+dependent variable. Linear regressions are popular among ML projects used
+for predictive analysis.
+"""
+
+from torch import nn
```
Comment (albanD) [inline beginner_source/linear_regression_tutorial.py:12]: This can be cleaned up to
```python
import torch
from torch import nn, tensor
```

File: beginner_source/linear_regression_tutorial.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+    # Backward pass, update optimizer parameter given current gradient.
+    loss.backward()
+    optimizer.step()
+
+
+# After training, test with a novel x_value
+x = tensor([[5.0]])
+y_pred = model(x)
+print("Prediction:", y_pred.data[0][0])
```
Comment (albanD) [inline beginner_source/linear_regression_tutorial.py:75]: Don't use `.data`, you can use `.detach()`.

**Comment by svekars** [comment]
Closing for now as we have not heard from the original author in a while. Feel free to reopen, if needed.
